library/src/main/kotlin/org/codice/compliance/SAMLComplianceException.kt

```diff
@@ -49,17 +49,21 @@ class SAMLComplianceException : Exception {
         }
 
         @Suppress("LongParameterList")
-        fun createWithPropertyMessage(code: SAMLSpecRefMessage,
+        fun createWithPropertyMessage(vararg codes: SAMLSpecRefMessage,
                                       property: String,
                                       actual: String?,
                                       expected: String? = null,
                                       node: Node? = null): SAMLComplianceException {
+            val samlExceptions = codes.map(Companion::readCode)
+                    .fold("SAML Specification References:\n") { acc, s ->
+                        "$acc\n$s"
+                    }
```

Comment:
Skipping over this because I'm in a rush and it takes me a second to wrap my head around the fold thing 😉

Reply:
This was a good resource http://kotlination.com/kotlin/kotlin-fold-example-list-map.

From my understanding, the `.fold` method is like a `forEach` that keeps appending/adding to a single value.
So in this case, `"SAML Specification References:\n"` is the initial value that is stored in the lambda variable `acc`. The other lamba variable `s` is the current element of the map.
The body of the lamba expresstion `"$acc\n$s"` is saying to take the `acc` variable, and append to it `\n' and `s`, then store that back into `acc` for the next iteration.